rpc/src/test/scala/org/bitcoins/rpc/RpcClientTest.scala

```diff
@@ -0,0 +1,517 @@
+
+class RpcClientTest extends AsyncFlatSpec {
+  implicit val system = ActorSystem()
+  implicit val m = ActorMaterializer()
+  implicit val ec = m.executionContext
+  implicit val regTestNetworkParam = RegTest
+
+  val client = new RpcClient
+  val logger = BitcoinSLogger.logger
+
+  behavior of "RpcClient"
+
+  it should "be able to get blockchain info" in {
+    client.getBlockChainInfo.flatMap { info =>
+      logger.info(info.toString)
+      assert(info.chain == "regtest")
+      client.getBestBlockHash.map(bestHash =>
+        assert(info.bestblockhash == bestHash))
+      assert(info.softforks.length >= 3)
+      assert(info.bip9_softforks.keySet.size >= 2)
+    }
+  }
+
+  it should "be able to create a raw transaction" in {
+    client.generate(2).flatMap { blocks =>
+      client.getBlock(blocks(0)).flatMap { block0 =>
+        client.getBlock(blocks(1)).flatMap { block1 =>
+          client.getTransaction(block0.tx(0)).flatMap { transaction0 =>
+            client.getTransaction(block1.tx(0)).flatMap { transaction1 =>
+              val input0 =
+                TransactionOutPoint(transaction0.txid,
+                                    UInt32(transaction0.blockindex.get))
+              val input1 =
+                TransactionOutPoint(transaction1.txid,
+                                    UInt32(transaction1.blockindex.get))
+              val sig: ScriptSignature = ScriptSignature.empty
+              client.getNewAddress().flatMap { address =>
+                client
+                  .createRawTransaction(
+                    Vector(TransactionInput(input0, sig, UInt32(1)),
+                           TransactionInput(input1, sig, UInt32(2))),
+                    Map((address.value, Bitcoins(1))))
+                  .map { transaction =>
+                    logger.info(transaction.toString)
+                    assert(transaction.inputs(0).sequence == UInt32(1))
+                    assert(transaction.inputs(1).sequence == UInt32(2))
+                    assert(
+                      transaction
+                        .inputs(0)
+                        .previousOutput
+                        .txId
+                        .flip == input0.txId)
+                    assert(
+                      transaction
+                        .inputs(1)
+                        .previousOutput
+                        .txId
+                        .flip == input1.txId)
+                  }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private def createRawTransaction: Future[Transaction] = {
+    client.generate(2).flatMap { blocks =>
+      client.getBlock(blocks(0)).flatMap { block0 =>
+        client.getBlock(blocks(1)).flatMap { block1 =>
+          client.getTransaction(block0.tx(0)).flatMap { transaction0 =>
+            client.getTransaction(block1.tx(0)).flatMap { transaction1 =>
+              val input0 =
+                TransactionOutPoint(transaction0.txid,
```

Comment:
the `.txid` here is big endian not little endian. We need to be careful with these. Probably should just add a new hash type to be `BEDoubleSha256Digest` to distinguish the endianness